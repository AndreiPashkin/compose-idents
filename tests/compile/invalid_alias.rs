use compose_idents::compose_idents;

compose_idents!(alias = lower(42), {
    fn alias() -> u32 {
        42
    }
});

fn main() {
    assert_eq!(qux_foo_baz(), 42);
}
