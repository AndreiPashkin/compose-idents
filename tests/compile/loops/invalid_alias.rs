//! Tests that invalid aliases produce correct compile-errors.
use compose_idents::compose_idents;

compose_idents!(
    for suffix in [foo, 1]

    my_fn = concat(my, _, suffix),
    {
        fn my_fn() -> u32 {
            42
        }
    }
);

fn main() {
    assert_eq!(my_foo(), 42);
    assert_eq!(my_bar(), 42);
}
