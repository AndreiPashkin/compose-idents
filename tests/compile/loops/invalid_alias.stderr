error: Invalid alias value: 1
 --> tests/compile/loops/invalid_alias.rs:5:25
  |
5 |     for suffix in [foo, 1]
  |                         ^

error[E0425]: cannot find function `my_foo` in this scope
  --> tests/compile/loops/invalid_alias.rs:16:16
   |
16 |     assert_eq!(my_foo(), 42);
   |                ^^^^^^ not found in this scope

error[E0425]: cannot find function `my_bar` in this scope
  --> tests/compile/loops/invalid_alias.rs:17:16
   |
17 |     assert_eq!(my_bar(), 42);
   |                ^^^^^^ not found in this scope
